src/pages/zh-cn/reference/api-reference.md

```diff
@@ -122,21 +169,75 @@ Astro.response.headers.set('Set-Cookie', 'a=b; Path=/;');
 
 ### `Astro.canonicalURL`
 
-当前页面的[规范链接][canonical]。如果设置了 `site` 选项，网站的源将是该链接的源。
+:::caution[废弃]
+使用 [`Astro.url`](#astrourl) 来构建你自己的标准链接。
+:::
+
+当前页面的[标准链接][canonical]。
+
+### `Astro.clientAddress`
 
-你也可以使用 `canonicalURL` 来获取当前页面的 `pathname`。
+<Since v="1.0.0-rc" />
+
+指定请求的 [IP 地址](https://en.wikipedia.org/wiki/IP_address)。这个属性只在为 SSR（服务器端渲染）构建时可用，不应该用于静态网站。
 
 ```astro
 ---
-const path = Astro.canonicalURL.pathname;
+const ip = Astro.clientAddress;
 ---
 
-<h1>Welcome to {path}</h1>
+<div>你的 IP 地址是：<span class="address">{ ip }</span></div>
+```
+
+### `Astro.url`
+
+<Since v="1.0.0-rc" />
+
+[URL](https://developer.mozilla.org/en-US/docs/Web/API/URL )对象，由当前 `Astro.request.url` 的链接字符串值构成。对于与请求的链接的个别属性进行交互是有用的，如路径名和起源。
+
+相当于做 `new URL(Astro.request.url)`。
+
+```astro
+<h1>当前链接是：{Astro.url}</h1>
+<h1>当前链接路径名是：{Astro.url.pathname}</h1>
+<h1>当前链接源是：{Astro.url.origin}</h1>
+```
+
+你也可以使用 `Astro.url` 来创建新的链接，并把它作为参数传给 [`new URL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)。
+
+```astro
+---
+// 示例：使用你的生产域名构建一个规范的URL
+const canonicalURL = new URL(Astro.url.pathname, Astro.site);
+// 示例：使用你目前的域名构建一个用于SEO元标签的URL
+const socialImageURL = new URL('/images/preview.png', Astro.url);
+---
+<link rel="canonical" href={canonicalURL} />
+<meta property="og:image" content={socialImageURL} />
 ```
 
 ### `Astro.site`
 
-`Astro.site` 返回根据 Astro 配置中的 `.site` 生成的 `URL`。如果没指定，将返回根据 `localhost` 生成的链接。
+`Astro.site` 返回根据 Astro 配置中的 `site` 生成的 `URL`。如果没指定，将返回根据 `localhost` 生成的链接。
+
+### `Astro.generator`
+
+<Since v="1.0.0" />
+
+`Astro.generator` 是个便捷方法，它可以添加与你当前 Astro 版本一致的 [`<meta name="generator">`](https://html.spec.whatwg.org/multipage/semantics.html#meta-generator) 标签。它遵循的格式是 `"Astro v1.x.x"`。
+
+```astro mark="Astro.generator"
+<html>
+  <head>
+    <meta name="generator" content={Astro.generator} />
+  </head>
+  <body>
+    <footer>
+      <p>Built with <a href="https://astro.build">{Astro.generator}</a></p>
```

Comment:
Not essential, but do you think we should translate “Built with” in this code sample?